packages/samples/headless-react/src/components/url-manager/url-manager.ts

```diff
@@ -21,6 +26,19 @@ export function bindUrlManager(engine: SearchEngine) {
   window.addEventListener('hashchange', onHashChange);
   const unsubscribeManager = urlManager.subscribe(() => {
     const hash = `#${urlManager.state.fragment}`;
+
+    if (!statusControllers.state.firstSearchExecuted) {
+      // The purpose of using `replaceState()` instead of `pushState()` in this case is to ensure that the URL reflects the current state of the search page on the first interface load.
+
+      // If `pushState()` were used instead, you could possibly enter in a history loop, which could result in having to click the back button multiple times without being able to  return to a previous page.
+      // This situation happens when you have a component like the Tab component that adds a new state to the browser history stack.
+
+      // Replacing the current state of the browser history with a new state, effectively updating the URL without adding a new entry to the history stack.
+      // For more info, visit https://docs.coveo.com/en/headless/latest/usage/synchronize-search-parameters-with-the-url/
```

Comment:
```suggestion
      // If `pushState()` were used instead, users could possibly enter in a history loop, having to click the back button multiple times without being able to return to a previous page.
      // This situation happens with components such as the Tab component, which adds a new state to the browser history stack.
      // This replaces the current state of the browser history with a new state, effectively updating the URL without adding a new entry to the history stack.
      // See https://docs.coveo.com/en/headless/latest/usage/synchronize-search-parameters-with-the-url/
```

Reply:
I'll add details to the handwritten docs too https://docs.coveo.com/en/headless/latest/usage/synchronize-search-parameters-with-the-url/ 👍